runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/config/EnvironmentSetting.kt

```diff
@@ -41,9 +42,10 @@ public data class EnvironmentSetting<T>(
  * property first, falling back to the environment variable if necessary. If neither is set, it returns the setting's
  * default value.
  * @param platform The [PlatformEnvironProvider] to use for reading system properties and environment variables.
+ * Defaults to [PlatformProvider.System].
  */
 @InternalApi
-public fun <T> EnvironmentSetting<T>.resolve(platform: PlatformEnvironProvider): T? {
+public fun <T> EnvironmentSetting<T>.resolve(platform: PlatformEnvironProvider = PlatformProvider.System): T? {
```

Comment:
Nice change here, in my LogMode PR I passed `PlatformProvider.System` but it's better to add a default